lib/chatops/controller.rb

```diff
@@ -118,26 +118,30 @@ def ensure_user_given
     end
 
     def ensure_chatops_authenticated
-      body = request.raw_post || ""
-      signature_string = [@chatops_url, @chatops_nonce, @chatops_timestamp, body].join("\n")
-      # We return this just to aid client debugging.
-      response.headers["Chatops-Signature-String"] = Base64.strict_encode64(signature_string)
       raise ConfigurationError.new("You need to add a client's public key in .pem format via #{Chatops.public_key_env_var_name}") unless Chatops.public_key.present?
-      if signature_valid?(Chatops.public_key, @chatops_signature, signature_string) ||
-          signature_valid?(Chatops.alt_public_key, @chatops_signature, signature_string)
-          return true
+
+      body = request.raw_post || ""
+
+      valid = @chatops_urls.any? do |url|
+        signature_string = [url, @chatops_nonce, @chatops_timestamp, body].join("\n")
+        # We return this just to aid client debugging.
+        response.headers["Chatops-Signature-String"] = Base64.strict_encode64(signature_string)
+        if signature_valid?(Chatops.public_key, @chatops_signature, signature_string) ||
+            signature_valid?(Chatops.alt_public_key, @chatops_signature, signature_string)
+            return true
```

Comment:
should this be `next true` or we could remove `return true if valid`?

Reply:
this is a block for `.any?` method, so `return true` looks valid here

Reply:
I think `return` will exit the function

Reply:
<img width="280" alt="image" src="https://github.com/github/chatops-controller/assets/6463364/fccd39df-18ec-497d-8b24-2122d65db420">

<img width="288" alt="image" src="https://github.com/github/chatops-controller/assets/6463364/f784a301-6ac9-40b3-a7eb-9d8b277b5d67">

Reply:
from the block, yes

Reply:
so `valid` will be never true, that's my point

Reply:
Fixed ❤️